improver_tests/utilities/test_enforce_consistency.py

```diff
@@ -0,0 +1,193 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for enforce_consistency utilities."""
+
+import numpy as np
+import pytest
+from iris.cube import Cube
+
+from improver.synthetic_data.set_up_test_cubes import (
+    set_up_percentile_cube,
+    set_up_probability_cube,
+)
+from improver.utilities.enforce_consistency import EnforceConsistentForecasts
+
+LOCAL_MANDATORY_ATTRIBUTES = {
+    "title": "Unit Test Data",
+    "source": "Unit Test",
+    "institution": "Met Office",
+}
+
+
+def get_percentiles(value, shape):
+    """
+    Correctly shape the value provided into the desired shape. The value is used as the
+    middle of three percentiles. The value is reduced or increased by 10 to create the
+    bounding percentiles.
+    """
+    return np.broadcast_to(
+        np.expand_dims([value - 10, value, value + 10], axis=(1, 2),), shape,
+    )
+
+
+def get_percentile_forecast(value, shape, name):
+    """
+    Create a percentile forecast cube.
+    """
+    data = get_percentiles(value, shape)
+    forecast_cube = set_up_percentile_cube(
+        np.full(shape, data, dtype=np.float32), [10, 50, 90], name=name, units="m s-1",
+    )
+    return forecast_cube
+
+
+def get_probability_forecast(value, shape, name) -> Cube:
+    """
+    Create a probability forecast cube.
+    """
+    data = np.full(shape, fill_value=value, dtype=np.float32)
+    forecast_cube = set_up_probability_cube(
+        data,
+        variable_name=name,
+        thresholds=[0.3125, 0.5, 0.6875],
+        attributes=LOCAL_MANDATORY_ATTRIBUTES,
+    )
+    return forecast_cube
+
+
+def get_expected(forecast_data, bound_data, comparison_operator):
+    """
+    Calculate the expected result via a different method to that used within the plugin.
+    """
+    diff = bound_data - forecast_data
+    if comparison_operator == ">=":
+        diff = np.clip(diff, 0, None)
+    else:
+        diff = np.clip(diff, None, 0)
+    expected = forecast_data + diff
+    return expected
+
+
+@pytest.mark.parametrize("forecast_type", ("percentile", "probability"))
+@pytest.mark.parametrize("additive_amount", (-15, 0, 15))
```

Comment:
There are 2 issues here:

I made a mistake within the test, for probability forecasts the `additive_amount` should be divided by 100. If I had defined this correctly, the smallest value achieved by the bounding values would have been 0 (`= -15 + (0.5*30)`) and similarly for probability data. 

I've now changed the `additive_amount` to take the values `(-12.5, 0, 12.5)`, so that the smallest value used is 2.5 for percentiles and 0.025 for probabilities.

Regarding whether to check for implausible values in the output, I err on the side of leaving the user in control of their own fate here. It's simple enough to check that all probabilities are in `[0, 1]`, but once we start considering reasonable values for percentiles/realisations this will vary on a diagnostic to diagnostic basis.

Reply:
Discussed in stand-up. Decided to limit the functions available to be applied to probability forecasts to only the identity function